api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxNewCohortQueries.java

```diff
@@ -237,4 +241,381 @@ public CohortDefinition getTxNewCompositionCohort(String cohortName) {
+    map.put("18", hivMetadata.getARVPharmaciaEncounterType().getEncounterTypeId());
+    map.put("1256", hivMetadata.getStartDrugs().getConceptId());
+    map.put("2", hivMetadata.getARTProgram().getProgramId());
+
+    CommonQueries commonQueries = new CommonQueries(new CommonMetadata(), new HivMetadata());
+
+    String query =
+        "       SELECT patient_id "
+            + " FROM ( "
+            + commonQueries.getARTStartDate(true)
+            + "       ) start "
+            + " WHERE start.first_pickup <= '2023-12-21' ";
+
+    StringSubstitutor sb = new StringSubstitutor(map);
+    cd.setQuery(sb.replace(query));
+    return cd;
+  }
+
+  /**
+   * All patients whose earliest ART start date (NEW_FR4.1) falls on or after (>=) 21 December
+   *
+   * <p>2023.
+   *
+   * <p>The system will identify the patient ART start date by selecting the earliest date from the
+   * following sources:
+   *
+   * <p>First ever drug pick up date registered on (FILA) Drug initiation date (ARV PLAN = START
+   * DRUGS) during the pharmacy or clinical visits First start drugs date set in in clinical tools
+   * (Ficha de Seguimento Adulto and Ficha de Seguimento Pediatria) or Ficha Resumo - Master Card.
+   * Date of enrollment in ART Program First ever drug pick-up date registered on (Recepção Levantou
+   * ARV) – Master Card
+   *
+   * <p>Note: Ensure that the ART start date is truly the first occurrence ever for the patient.
+   * This is particularly important for patients that have different ART start dates registered in
+   * the system.
+   */
+  public CohortDefinition getPatientsArtStartDate() {
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName("Patient’s earliest ART start date from pick-up and clinical sources");
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+
+    Map<String, Integer> map = new HashMap<>();
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+    map.put("9", hivMetadata.getPediatriaSeguimentoEncounterType().getEncounterTypeId());
+    map.put("53", hivMetadata.getMasterCardEncounterType().getEncounterTypeId());
+    map.put("52", hivMetadata.getMasterCardDrugPickupEncounterType().getEncounterTypeId());
+    map.put("23866", hivMetadata.getArtDatePickupMasterCard().getConceptId());
+    map.put("23865", hivMetadata.getArtPickupConcept().getConceptId());
+    map.put("1065", hivMetadata.getPatientFoundYesConcept().getConceptId());
+    map.put("1190", hivMetadata.getARVStartDateConcept().getConceptId());
+    map.put("1255", hivMetadata.getARVPlanConcept().getConceptId());
+    map.put("18", hivMetadata.getARVPharmaciaEncounterType().getEncounterTypeId());
+    map.put("1256", hivMetadata.getStartDrugs().getConceptId());
+    map.put("2", hivMetadata.getARTProgram().getProgramId());
```

Comment:
1. No need to map the query. It was already mapped on common query class
2. You never check if the Art Start Date is after report start date

Reply:
Resolved